heareval/tasks/pipeline.py

```diff
@@ -550,12 +677,26 @@ def requires(self):
         }
 
     def run(self):
-        for audiofile in tqdm(list(self.requires()["corpus"].workdir.glob("*.wav"))):
-            # Compare the filename with the slug.
-            # Note that the slug does not have the extension of the file
-            split = self.metadata.loc[
-                self.metadata["slug"] == audiofile.stem, "split"
-            ].values[0]
+        audiofiles = set(self.requires()["corpus"].workdir.glob("*.wav"))
+        unique_filestems = set(
+            self.metadata["unique_filestem"].drop_duplicates().values
+        )
+        # Gross, let's never do this again
+        if self.task_config["version"].split("-")[-1] == "small":
+            # Many filestems in the metadata won't be in the
+            # small corpus as audio.
+            assert len(audiofiles) <= len(unique_filestems)
+        else:
+            assert len(audiofiles) == len(unique_filestems)
```

Comment:
This assertion is still failing. During SubsampleSplits we select only a portion of the samples from metadata, and then symlink them to a new folder. `len(unique_filestems)` is all the audio files before subsampling. `len(audiofiles)` is all the audio files *after* subsampling. If subsampling is applied this assertion will never hold. i.e for both nsynth and speech commands.

Reply:
We have a couple options here: 1) don't make this assertion, or 2) remove the data from files that don't make it into the subsample from metadata.

Reply:
Looks like option 1 was already taken in #30